filter/set.go

```diff
@@ -0,0 +1,67 @@
+package filter
+
+import (
+	"sync"
+)
+
+// set is a Filter that contains 2 or more Filters
+type set struct {
+	filters []Filter
+}
+
+// NewSet returns a set of Filters
+func NewSet(filters ...Filter) Filter {
+	return &set{filters: filters}
+}
+
+// Run against an input and output channel until the input channel is closed.
+// The output channel is closed on return.
+func (s *set) Run(in <-chan int, out chan<- int) {
+	var wg sync.WaitGroup
+
+	if len(s.filters) <= 1 {
+		wg.Add(1)
+		if len(s.filters) == 0 {
+			// if there are no filters, create a no operations filter to connect in and out.
+			s.filters = append(s.filters, NewNOP())
+		}
+		go func() {
+			s.filters[0].Run(in, out)
+			wg.Done()
+		}()
+		wg.Wait()
+		return
+	}
+
+	wg.Add(len(s.filters))
+	// Create a list for all channels
+	channels := make([]chan int, len(s.filters)-1)
+	channels[0] = make(chan int)
```

Comment:
You already have two known channels:

```suggestion
	channels[0] = in
	channels[len(filters)] = out
```

This does require a check a bit later:

```go
// Create the output channel if required
if channels[i+1] == nil {
	channels[i+1] = make(chan int)
}
```